src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -27,4 +28,34 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null) {
+            return false;
+        }
+        if (obj.getClass().equals(Rectangle.class)) {
+            Rectangle rectangle = (Rectangle) obj;
+            return (color != null && color.equals(rectangle.color)
+                    || (rectangle.color == null && color == null))
+                    && (width != null && width.equals(rectangle.width)
+                    || (rectangle.width == null && width == null))
+                    && (length != null && length.equals(rectangle.length)
+                    || (rectangle.length == null && length == null));
+        }
+        return false;
+    }
+
+    @Override
+    public int hashCode() {
+        int result = 0;
+        result = width != null ? HASHCODENUMBER * result + width.hashCode() : HASHCODENUMBER;
+        result = length != null ? HASHCODENUMBER * result + length.hashCode()
+                : HASHCODENUMBER * result;
+        result = color != null ? HASHCODENUMBER * result + color.hashCode() : HASHCODENUMBER % 10;
```

Comment:
what happens when we multiply by 0 on line 55?
also, let's use the same calculation logic for all the fields

Reply:
Done)